### ISSUE TITLE
[ENH] benchmarks such as `ForecastingBenchmark` do not test all estimators on the same folds but should

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#10881]
The user expectation is that multiple forecasters, classifiers, etc , in a benchmarking experiment are run on the same folds.

However, the folds are re-sampled per estimator, which for non-deterministic splitters will result in different folds.

This is also problematic because most post-hoc analysis methods (such as Friedman test or Critical Differences) assume paired/blocked samples, obtained on the same folds.

For `ForecastingBenchmark`, this is not much of a problem, since the common splitters (if not all, in `sktime`), are deterministic. However, also here, for user-made splitters, the problem occurs.

This was noticed by @yash-sangwan in #10818, and solved by a "fixed splitter" wrapper for grid search

The solution is a hack though - I think the correct solution should live either in `evaluate` (accepting multiple estimators), or in the benchmarks. Some thoughts:

* I think this should be an option that the user can set. The default should be "same splits for all estimators".
* separately, we should reevaluate whether the grid search algorithms should use `evaluate` - or, whether they should use `ClassificationBenchmark` etc.

FYI @jgyasu.

---

Internal issue, not a good first issue for new contributors.

**Comment by jgyasu** [comment]
Isn't this behaviour expected by virtue of the current design of the framework? At least to the developers it is. `evaluate` is a backtesting tool and its scope is limited to evaluating a single estimator, it is what is responsible for fold generation, the `ForecastingBenchmark`, `ClassificationBenchmark` etc. are responsible for orchestrating the entire experiment and calls `evaluate` per estimator. I am not sure of the initial design decisions that led to using `evaluate` as the atomic evaluation units in the benchmark experiments, but if we want to achieve this, we will have to make the benchmarks independent of the `evaluate` and implement the evaluation logic in benchmark classes themselves, so my questions is if we want to rework and go in that direction?

**Comment by jgyasu** [comment]
> This is also problematic because most post-hoc analysis methods (such as Friedman test or Critical Differences) assume paired/blocked samples, obtained on the same folds.

It should not be a problem for the deterministic splitters, but yes I do see how it can cause problems for other cases.

**Comment by mateenali66** [comment]
Even in the deterministic-splitter case the paired assumption is a property of the current splitter implementations, not something the framework asserts. It holds silently or breaks silently, and a user-made splitter breaks it silently today.

A middle ground that keeps `evaluate` as the atomic unit: each `evaluate` call emits a fold fingerprint (a hash of the train/test index arrays per fold) into its results, and the benchmark verifies the fingerprints match across estimators before any Friedman/CD step, erroring or falling back to unpaired tests on mismatch. The "same splits for all estimators" default then becomes a checked property rather than a trusted one, so the rework question in the first comment can be deferred without shipping invalid CD diagrams in the meantime.

**Comment by yash-sangwan** [comment]
Minimal example for the fold issue.

evaluate re-samples folds on every call if the splitter shuffles.
Same estimator, same data, same splitter, different scores.

```python
import warnings
from sklearn.model_selection import KFold
from sktime.classification.dummy import DummyClassifier
from sktime.classification.model_evaluation import evaluate
from sktime.datasets import load_unit_test

warnings.filterwarnings("ignore")

X, y = load_unit_test(split="train")
clf = DummyClassifier(strategy="most_frequent")
cv = KFold(n_splits=3, shuffle=True)

print("evaluate #1:", evaluate(clf, cv=cv, X=X, y=y)["test_accuracy_score"].round(3).tolist())
print("evaluate #2:", evaluate(clf, cv=cv, X=X, y=y)["test_accuracy_score"].round(3).tolist())

**Comment by mateenali66** [comment]
@yash-sangwan that repro is exactly the case the fingerprint catches. Two `evaluate` calls, same splitter, different folds, and nothing in the results says so.

First half is up as #11024: a `fold_fingerprint` column on `evaluate` for forecasting, classification and regression, hashing the train/test indices only, computed before fit so a failed fold still carries it. The benchmark-side comparison across estimators, before any Friedman or CD step, is the second half and is not in that PR yet.
